src/qibosoq/programs/base.py

```diff
@@ -74,6 +73,9 @@ def __init__(
         # pylint: disable-next=too-many-function-args
         super().__init__(soc, asdict(qpcfg))
 
+        self.reps = qpcfg.reps  # must be done after QickProgram init
```

Comment:
If I remember correctly, the problem was for `self.soft_avgs` because of this:
https://github.com/openquantumhardware/qick/blob/6c823b348e9c49a5db3b35b98c698cf874218c4f/qick_lib/qick/averager_program.py#L31-L34

Reply:
Ok, I see. But this class is not inheriting (not even indirectly) from the `AveragerProgram`.

Maybe we should make a `qibosoq.AveragerProgram` (even with a different name), or directly define these attributes in `ExecutePulseSequence` and `ExecuteSweep`.